python/tank/platform/application.py

```diff
@@ -145,57 +181,77 @@ def destroy_app(self):
         pass
     
     ##########################################################################################
-    # logging methods, delegated to the current engine
+    # logging methods
 
     def log_debug(self, msg):
         """
         Logs a debug message.
 
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
+
         :param msg: Message to log.
         """
-        self.engine.log_debug(msg)
+        if self.engine.supports_018_logging:
+            self.log.debug(msg)
+        else:
+            self.engine.log_debug(msg)
 
     def log_info(self, msg):
         """
         Logs an info message.
 
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
+
         :param msg: Message to log.
         """
-        self.engine.log_info(msg)
+        if self.engine.supports_018_logging:
+            self.log.info(msg)
+        else:
+            self.engine.log_info(msg)
 
     def log_warning(self, msg):
         """
-        Logs a warning message.
+        Logs an warning message.
+
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
 
         :param msg: Message to log.
         """
-        self.engine.log_warning(msg)
+        if self.engine.supports_018_logging:
+            self.log.warning(msg)
+        else:
+            self.engine.log_warning(msg)
 
     def log_error(self, msg):
         """
         Logs an error message.
 
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
+
         :param msg: Message to log.
         """
-        self.engine.log_error(msg)
+        if self.engine.supports_018_logging:
+            self.log.error(msg)
+        else:
+            self.engine.log_error(msg)
 
     def log_exception(self, msg):
         """
-        Logs an exception.
-
-        This will contain a full traceback and is typically called from
-        within an exception handler::
-
-            try:
-                do_stuff()
-            except Exception:
-                self.log_exception("A general error was raised")
+        Logs an exception message.
 
-        The message will be emitted as an error message.
+        .. deprecated:: 0.18
+            Use :meth:`Engine.log` instead.
 
         :param msg: Message to log.
         """
-        self.engine.log_exception(msg)
+        if self.engine.supports_018_logging:
```

Comment:
Could you explain why this if is necessary? Either the derived engine reimplemented the method, in which case calling `engine.log_exceptio`n will log  through the engine's own logging system or it hasn't, in which case the default implementation should call `self.log_exception`.